include/fiction/algorithms/simulation_sidb/distance_matrix.hpp

```diff
@@ -0,0 +1,50 @@
+#ifndef FICTION_DISTANCE_MATRIX_HPP
+#define FICTION_DISTANCE_MATRIX_HPP
+//
+// Created by Jan Drewniok on 07.12.22.
+//
+
+#include "fiction/algorithms/path_finding/distance.hpp"
+#include "fiction/algorithms/simulation_sidb/constants.hpp"
+#include "fiction/layouts/coordinates.hpp"
+#include "fiction/technology/charge_distribution_surface.hpp"
+#include "fiction/traits.hpp"
+#include "fiction/types.hpp"
+
+namespace fiction
+{
+/**
+ * The distance matrix is an unordered map with pairs of cells as key and the corresponding euclidean distance as value.
+ */
+template <typename Lyt, typename Dist = double>
+using distance_matrix = std::unordered_map<std::pair<const cell<Lyt>, const cell<Lyt>>, Dist>;
+
+/**
+ * The distance matrix stores all euclidean inter-distances. The euclidean distance between
```

Comment:
This docstring fits the `distance_matrix` type better than this function. It's unclear what it does based on the docstring